operator/src/main/java/org/keycloak/operator/controllers/KeycloakUpdateJobDependentResource.java

```diff
@@ -127,56 +123,66 @@ private void addPodSpecTemplate(JobSpecFluent<?> builder, Keycloak keycloak, Con
     }
 
     private PodSpec createPodSpec(Context<Keycloak> context) {
-        var allVolumes = getAllVolumes(context);
-        Collection<String> requiredVolumes = new HashSet<>();
-        var builder = new PodSpecBuilder();
+        StatefulSet current = ContextUtils.getCurrentStatefulSet(context).orElseThrow();
+        StatefulSet desired = ContextUtils.getDesiredStatefulSet(context);
+
+        // start off with the desired statefulset state
+        var builder = desired.getSpec().getTemplate().getSpec().edit();
         builder.withRestartPolicy("Never");
-        addInitContainer(builder, context, allVolumes.keySet(), requiredVolumes);
-        addContainer(builder, context, allVolumes.keySet(), requiredVolumes);
+
+        // remove things we don't want
+
+        // TODO: this may actually not be correct - via the unsupported PodTemplate someone could make configuration dependent upon an init container
+        builder.withInitContainers();
+        builder.withContainers();
+
+        // TODO: should the scheduling fields be tweaked
+
+        // mix in the existing state
+        var currentSecrets = builder.buildImagePullSecrets();
+        current.getSpec().getTemplate().getSpec().getImagePullSecrets().stream().filter(s -> !currentSecrets.contains(s)).forEach(builder::addToImagePullSecrets);
+        // TODO: if the name is the same, but the volume has changed the behavior will be inconsistent
```

Comment:
I'd say if we would make the volume name a bit more specific we should be good for now. 

See above for a suggestion.

Reply:
This concerns is about a difference between the desired and existing state. Lets say in the existing state you have a volume name config pointed at configmap X, and in the desired state you a volume also named config pointed at configmap Y - we are not accounting for that possibility.

This could happen with more than just via the unsupported podTemplate, such as via the cache config configmap.

If we want to account for this we'll need to do a name remapping as part of the merge.

Reply:
Thank you, I now see your point. I'd say this is in the unsupported space, and within the margin of error when we accept it as-is now.